src/Repositories/DeleteOldExceptionCommandRepository.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Taecontrol\MoonGuard\Repositories;
+
+use Taecontrol\MoonGuard\Models\ExceptionLog;
+
+class DeleteOldExceptionCommandRepository
+{
+    protected static string $modelClassConfigKey = 'moonguard.exceptions.exception_log.model';
+
+    public static function isEnabled(): bool
+    {
+        return config('moonguard.exception_deletion.enabled');
+    }
+
+    public static function getExceptionAge(): int
+    {
+        return config('moonguard.exception_deletion.delete_exceptions_older_than_minutes');
+    }
+
+    public static function deleteOldExceptions(int $time): void
+    {
+        $exceptions = ExceptionLog::where('thrown_at', '<', now()->subMinutes($time))->get();
+        $exceptions->each(function ($exception) {
+            $exception->delete();
+        });
+    }
```

Comment:
```suggestion
    public static function deleteOldExceptions(int $time): void
    {
        $exceptions = ExceptionLogGroupRepository::query()
          ->where('thrown_at', '<', now()->subMinutes($time))
          ->get();
          
        $exceptions->delete();
    }
```

You must use the `ExceptionLogGroupRepository` in this case, we have to resolve which model to use and then you can use the `query()` method with `where()` to work normally with Eloquent.

Also you should not loop a collection of models to run queries. If we loop the `$exceptions` to run the `delete()` query you are making a query for each model. You should be able to just use `$exceptions->delete()`.